### DISCUSSION TITLE
Placing custom controls relative to keyboard

### DISCUSSION THREAD

**Discussion by Jaarson** [ionic-team/capacitor#6349] [Q&A] [answered]
Hey 👋
Is it possible to place custom controls (bar) relative to the keyboard, and then be able to communicate which button was tapped?
Only using custom native code perhaps?

![IMG_A7EFA3594D18-1](https://user-images.githubusercontent.com/16690523/222195545-0548b70c-91f7-4d62-b9e7-fb5d4365c198.jpeg)

**Comment by mlynch** [comment, marked as answer]
Yep, for the best experience it should be a native bar for perfect fixing to the keyboard, here's a capacitor plugin that shows how to do this: https://github.com/mlynch/capacitor-keyboard-toolbar-demo

**Reply by mlynch** [reply]
Wanted to add that the team over at Reflect took some of the basics of this plugin and extended it quite a bit, and they have a fantastic keyboard editor bar for their Capacitor app: https://reflect.app

**Reply by Jaarson** [reply]
Thanks! This is looking really good!
I'm currently checking if Capacitor is right for me, and seems it is 💪 
Some other things I'm also checking now: Supabase Auth and Realtime, and nice-to-haves: local SQLite, some sort of safe storage (iOS Keychain)
